bin/cloudgov/create-service-account

```diff
@@ -23,42 +36,44 @@ do
     keyname="$var"
   fi
 done
-resetkey=${1:-}
 
-CFV=$(cf -v)
-SWITCHBACK=0
-echo $CFV | grep 'version 7' -
-if [ $? != "0" ]; then
-  sudo apt-get remove cf8-cli && sudo apt-get install cf7-cli && cf version
-  if [ $? != "0" ]; then
-    echo Error occurred switching versions of CF CLI
-    exit 1
-  fi
-  SWITCHBACK=1
-fi
+#CFV=$(cf -v)
+#SWITCHBACK=0
+#echo $CFV | grep 'version 7' -
+#if [ $? != "0" ]; then
+#  sudo apt-get remove cf8-cli && sudo apt-get install cf7-cli && cf version
+#  if [ $? != "0" ]; then
+#    echo Error occurred switching versions of CF CLI
+#    exit 1
+#  fi
+#  SWITCHBACK=1
+#fi
 
-echo cf create-service cloud-gov-service-account space-deployer cci-service-account
-cf create-service cloud-gov-service-account space-deployer cci-service-account
+echo cf create-service cloud-gov-service-account space-deployer ${keyname}-service-account
+cf create-service cloud-gov-service-account space-deployer ${keyname}-service-account
 
-if [ "$SWITCHBACK" != "0" ]; then
-  sudo apt-get remove cf7-cli && sudo apt-get install cf8-cli && cf version
-  if [ $? != "0" ]; then
-    echo Error occurred switching versions of CF CLI
-    exit 1
-  fi
-fi
+#if [ "$SWITCHBACK" != "0" ]; then
+#  sudo apt-get remove cf7-cli && sudo apt-get install cf8-cli && cf version
+#  if [ $? != "0" ]; then
+#    echo Error occurred switching versions of CF CLI
+#    exit 1
+#  fi
+#fi
 
 if [ "$resetkey" = "--reset-key" ]; then
-  cf delete-service-key cci-service-account $keyname-service-key -f
+  cf delete-service-key ${keyname}-service-account ${keyname}-service-key -f
 fi
-cf create-service-key cci-service-account $keyname-service-key
-SERVICE_KEY=$(cf service-key cci-service-account $keyname-service-key | tail -n +3)
+cf create-service-key ${keyname}-service-account ${keyname}-service-key
+SERVICE_KEY=$(cf service-key ${keyname}-service-account ${keyname}-service-key | tail -n +3)
 echo $SERVICE_KEY
 SERVICE_USER=$( echo $SERVICE_KEY | jq -r '.credentials.username')
-cf set-space-role $SERVICE_USER $startorg dev SpaceDeveloper
-cf set-space-role $SERVICE_USER $startorg stage SpaceDeveloper
-cf set-space-role $SERVICE_USER $startorg prod SpaceDeveloper
-cf set-space-role $SERVICE_USER $startorg tools SpaceDeveloper
-cf set-space-role $SERVICE_USER $startorg shared-egress SpaceDeveloper
-cf set-space-role $SERVICE_USER $startorg shared-egress-dr SpaceDeveloper
-cf set-space-role $SERVICE_USER $startorg dev-dr SpaceDeveloper
+
+# you must already be logged in as an OrgManager to have enough permissions
+# to assign these cross-space developer roles to the service account's user
+
+cf set-space-role $SERVICE_USER $startorg dr SpaceDeveloper
+#cf set-space-role $SERVICE_USER $startorg dev SpaceDeveloper
```

Comment:
We'll want to un-comment these lines before merging.

Reply:
Should this just be setting the role for the service user within the current space, or all spaces?

It seems that since we run this script per space, we'd just want e.g. the dev user to have the role in the dev space?

Reply:
@akf - my latest commit will hopefully address the cross-space access needed for CCI.

Reply:
@arpage I like that ... for now.